libs/design-system/table/src/lib/table/table.component.html

```diff
@@ -30,10 +30,16 @@
               mat-sort-header
               [attr.data-column-type]="type"
               [sortActionDescription]="`Sort by ${column.label}`"
+              [class.vertical-divider]="verticalDividers() && column.column !== _columns()[0].column"
             >
               {{ column.label }}
             </th>
-            <td mat-cell *matCellDef="let element" [attr.data-column-type]="type">
+            <td
+              mat-cell
+              *matCellDef="let element"
+              [attr.data-column-type]="type"
+              [class.vertical-divider]="verticalDividers() && column.column !== _columns()[0].column"
```

Comment:
Remove `&& column.column !== _columns()[0].column` and move `[class.vertical-divider]="verticalDividers()"` to the `table` element. Then use a selector such as `.vertical-divider td:not(:first-child)` to apply the border